src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -193,12 +194,18 @@ private IServiceCollection BuildCommonServices(out AggregateException hostingSta
             if (!_options.PreventHostingStartup)
             {
                 var exceptions = new List<Exception>();
+                var assemblies = new HashSet<StringSegment>(StringSegmentComparer.OrdinalIgnoreCase);
 
                 // Execute the hosting startup assemblies
                 foreach (var assemblyName in _options.HostingStartupAssemblies)
                 {
                     try
                     {
+                        if (assemblies.Contains(assemblyName))
```

Comment:
Why even throw at all? Does it matter to anyone that it was skipped and not added twice?